### ISSUE TITLE
Variational MPO approach for finding NESS

### ISSUE THREAD

**Issue by 4psireal2** [shareloqs/MPSDynamics#8]
Many thanks for this great repo! Is there a possibility to find the non-equilibrium steady state using a variational approach? For [reference](https://arxiv.org/abs/1501.06786)

**Comment by tfmlaX** [comment]
Thank you! 
I know about this variational approach and it looks very interesting - we can look into it. Do you have some experience with this approach?

**Comment by 4psireal2** [comment]
Yes I do! Its implementation works for my system, the 1D quantum contact process. However, the ground state is the absorbing state and I want to probe NESS of non-zero particle number density instead. The hope was that NESS could be the first excited state but simulations with DMRG1(2) have only returned unphysical solutions. I think the problem could be due to the vectorised MPO ansatz which doesn't fulfill the positivity property of a density matrix. Also the excited states of Lˆ†L are typically different from the eigenstates of L so I wouldn't know how to interpret the result either.

I just thought with the MPS ansatz of your framework for open system, methods for closed system such as DMRG can be recycled.

**Comment by 4psireal2** [comment]
Some passsing thought:
1. How does the MPS's bond dimension grow with time using your implementation?
2. Is there some relation between the state found by long-time imaginary time evolution and the state found by long-time real time evolution? If yes, how?

**Comment by tfmlaX** [comment]
> I just thought with the MPS ansatz of your framework for open system, methods for closed system such as DMRG can be recycled.

This is non-trivial as we don't work at the level of the reduced systems dynamics but time-evolve the full state of the system and its environment. I'll have to take some time to understand a bit more deeply this variational approach to see how (if?) it can be reformulated/connected to what we do.

> How does the MPS's bond dimension grow with time using your implementation?

I would say this is model-dependent and I'm not sure that I'm able to give a general answer.

> Is there some relation between the state found by long-time imaginary time evolution and the state found by long-time real time evolution? If yes, how?

Very interesting question. Generally speaking, the long-time imaginary time evolution should bring the state towards the zero-temperature ground state; I do not know under which conditions it is the same as the steady state obtained with real time evolution.
